classes/Crons.php

```diff
@@ -427,47 +427,70 @@ public static function saveRepositoryFile($module, $project_id, $filename, $tabl
         if (($type == "" && $table_data != "" && $table_data['data'] != "" && $table_data['tooltip'] != "") || ($type == "graph" && !empty($table_data))) {
             #SAVE DATA IN FILE
             #create and save file with json
-            $storedName = date("YmdHis") . "_pid" . $project_id . "_" . ProjectData::getRandomIdentifier(6) . ".txt";
 
-            $file = fopen(EDOC_PATH . $storedName, "wb");
+            #Check if we have a different path than edocs
+            $path = $module->getProjectSetting('path',$project_id);
+            $storedName = $path == null ? date("YmdHis") . "_pid" . $project_id . "_" . ProjectData::getRandomIdentifier(6) . ".txt" : $filename;
+            $filePath = $path == null ? APP_PATH_TEMP . $storedName : $path . $storedName;
```

Comment:
This should likely be using getSafePath() to avoid vulnerabilities from a malicious user inserting sensitive system file paths in this project setting.  In general, getSafePath() should be used on all file paths.